File: python/raindrops/raindrops.py
def convert(number):
    result = str(number)

    if number % 3 == 0:
        result = "Pling"

    if number % 5 == 0:
        result += "Plang"

    if number % 7 == 0:
        result += "Plong"

    if not result.isdigit():
        result = result.replace(str(number), "")

    return result

File: php/raindrops/raindrops.php
<?php

define('FACTOR_SOUND_DICT', [3 => "Pling", 5 => "Plang", 7 => "Plong"]);

/**
 * Convert a number into a string that contains raindrop sounds corresponding to
 * certain potential factors.
 *
 * @param int $number Any integer
 *
 * @return string A string that contains raindrop sounds or given $number if no sound found.
 */
function raindrops(int $number): string
{
  $result = "";
  foreach (FACTOR_SOUND_DICT as $factor => $sound) {
    if ($number % $factor === 0) {
      $result .= $sound;
    }
  }

  return $result === "" ? $number : $result;
}
